src/sdk/clients/decorators/tokenPaymaster/getTokenPaymasterQuotes.ts

```diff
@@ -0,0 +1,154 @@
+  factoryData: `0x${string}` | undefined
+  callData: `0x${string}`
+  maxFeePerGas: string
+  maxPriorityFeePerGas: string
+  verificationGasLimit: string
+  callGasLimit: string
+  preVerificationGas: string
+  paymasterPostOpGasLimit: string
+  paymasterVerificationGasLimit: string
+}
+
+export type TokenPaymasterConfigParams = {
+  mode: PaymasterMode
+  sponsorshipInfo: {
+    smartAccountInfo: {
+      name: string
+      version: string
+    }
+  }
+  tokenInfo: {
+    tokenList: Address[]
+  }
+  expiryDuration: number
+  calculateGasLimits: boolean
+}
+
+/**
+ * Fetches paymaster quotes for ERC20 token payment options for a given UserOperation.
+ *
+ * @param userOp - The UserOperation to get paymaster quotes for
+ * @param client - Viem Client configured with BicoTokenPaymaster RPC methods
+ * @param tokenList - Array of ERC20 token addresses to get quotes for
+ *
+ * @returns Promise containing paymaster quotes response with fee details for each supported token
+ *
+ * @example
+ * ```typescript
+ * // Configure client with paymaster RPC
+ * const paymasterClient = createBicoPaymasterClient({
+ *     paymasterUrl
+ * })
+ *
+ * // Token addresses to get quotes for
+ * const tokenList = [
+ *   "0x...", // USDT
+ *   "0x..."  // USDC
+ * ];
+ *
+ * // Get paymaster quotes
+ * const quotes = await paymasterClient.getTokenPaymasterQuotes(userOp, tokenList);
+ *
+ * // Example response:
+ * // {
+ * //   mode: "ERC20",
+ * //   paymasterAddress: "0x...",
+ * //   feeQuotes: [{
+ * //     symbol: "USDT",
+ * //     decimal: 6,
+ * //     tokenAddress: "0x...",
+ * //     maxGasFee: 5000000,
+ * //     maxGasFeeUSD: 5,
+ * //     exchangeRate: 1,
+ * //     logoUrl: "https://...",
+ * //     premiumPercentage: "0.1",
+ * //     validUntil: 1234567890
+ * //   }],
+ * //   unsupportedTokens: []
+ * // }
+ * ```
+ */
+export const getTokenPaymasterQuotes = async (
+  userOp: UserOperation,
+  client: Client<
+    Transport,
+    Chain | undefined,
+    Account | undefined,
+    BicoTokenPaymasterRpcSchema
+  >,
+  tokenList: Address[]
+): Promise<TokenPaymasterQuotesResponse> => {
+  const quote = await client.request({
+    method: "pm_getFeeQuoteOrData",
+    params: [
+      {
+        sender: userOp.sender,
+        nonce: userOp.nonce.toString(),
+        factory: userOp.factory,
+        factoryData: userOp.factoryData,
+        callData: userOp.callData,
+        maxFeePerGas: userOp.maxFeePerGas.toString(),
+        maxPriorityFeePerGas: userOp.maxPriorityFeePerGas.toString(),
+        verificationGasLimit: BigInt(userOp.verificationGasLimit).toString(),
+        callGasLimit: BigInt(userOp.callGasLimit).toString(),
+        preVerificationGas: BigInt(userOp.preVerificationGas).toString(),
+        paymasterPostOpGasLimit:
+          userOp.paymasterPostOpGasLimit?.toString() ?? "0",
+        paymasterVerificationGasLimit:
+          userOp.paymasterVerificationGasLimit?.toString() ?? "0"
+      },
+      {
```

Comment:
This doesn't come from the paymasterContext in the account somewhere? Can't users specifiy the expiry for example?

Reply:
What is the issue with "paymasterVerificationGasLimit" ? You are pointing to this in this comment but you're mentioning something else. Users can set their own expiryDuration in the "paymasterContext", if they don't, the default values will be used.

Reply:
I was referring to the object starting on line 141. In here I might have expected that the expiryDuration might come from the end user, instead of it being hardcoded. 

Why is expiryDuration hardcoded in here? Can't you pull the paymasterContext out of the client here `client.paymasterContext` ?